test/beef_test.go

```diff
@@ -0,0 +1,172 @@
+package test
+
+import (
+	"context"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/bitcoin-sv/arc/pkg/api"
+	"github.com/bitcoin-sv/arc/pkg/metamorph/metamorph_api"
+	"github.com/libsv/go-bc"
+	"github.com/libsv/go-bt/v2"
+	"github.com/libsv/go-p2p/chaincfg/chainhash"
+	"github.com/stretchr/testify/require"
+)
+
+func TestBeef(t *testing.T) {
+	testCases := []struct {
+		name string
+	}{
+		{
+			name: "valid beef",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			arcClient, err := api.NewClientWithResponses(arcEndpoint)
+			require.NoError(t, err)
+
+			address, privateKey := getNewWalletAddress(t)
+
+			generate(t, 200)
+
+			txID := sendToAddress(t, address, 0.001)
+
+			hash := generate(t, 1)
+
+			rawTx := getRawTx(t, txID)
+			t.Logf("rawTx: %+v", rawTx)
+			require.Equal(t, hash, rawTx.BlockHash, "block hash mismatch")
+
+			blockData := getBlockDataByBlockHash(t, hash)
+			t.Logf("blockdata: %+v", blockData)
+
+			merkleHashes, txIndex := prepareMerkleHashesAndTxIndex(t, blockData.Txs, txID)
+			merkleTree := bc.BuildMerkleTreeStoreChainHash(merkleHashes)
+			bump, err := bc.NewBUMPFromMerkleTreeAndIndex(blockData.Height, merkleTree, txIndex)
+			require.NoError(t, err, "error creating BUMP from merkle hashes")
+
+			merkleRootFromBump, err := bump.CalculateRootGivenTxid(txID)
+			require.NoError(t, err, "error calculating merkle root from bump")
+			t.Logf("merkleroot from bump: %s", merkleRootFromBump)
+			require.Equal(t, blockData.MerkleRoot, merkleRootFromBump, "merkle roots mismatch")
+
+			utxos := getUtxos(t, address)
+			require.True(t, len(utxos) > 0, "No UTXOs available for the address")
+
+			newAddr, _ := getNewWalletAddress(t)
+
+			tx, err := createTx(privateKey, newAddr, utxos[0])
+			require.NoError(t, err, "could not create tx")
+			t.Logf("tx created, hex: %s", tx.String())
+
+			beef := buildBeef(t, rawTx.Hex, bump, tx)
+			t.Logf("beef created, hex: %s", beef)
+
+			body := api.POSTTransactionJSONRequestBody{
+				RawTx: beef,
+			}
+
+			expectedStatus := metamorph_api.Status_SEEN_ON_NETWORK
+
+			waitForStatus := api.WaitForStatus(expectedStatus)
+			params := &api.POSTTransactionParams{
+				XWaitForStatus: &waitForStatus,
+			}
+
+			response, err := arcClient.POSTTransactionWithResponse(context.Background(), params, body)
+			require.NoError(t, err)
+			require.Equal(t, http.StatusOK, response.StatusCode())
+			require.NotNil(t, response.JSON200)
+			require.Equal(t, expectedStatus.String(), response.JSON200.TxStatus, "status not SEEN_ON_NETWORK")
+
+			generate(t, 10)
+			t.Log("waiting for 15s to give ARC time to perform the status update on DB")
```

Comment:
Is it really a part of BEEF testing? Is SEEN_ON_NETWORK not enough?

Reply:
@arkadiuszos4chain testing the whole flow of the tx until it is MINED happens already in other e2e tests, but IMO it surely doesn't hurt to expect the tx status reaches MINED here another time